pkg/agent/plugin/workloadattestor/k8s/k8s.go

```diff
@@ -93,59 +102,115 @@ func (p *k8sPlugin) Attest(req *workloadattestor.AttestRequest) (*workloadattest
 		return &resp, nil
 	}
 
-	httpResp, err := p.httpClient.Get(fmt.Sprintf("http://localhost:%v/pods", p.kubeletReadOnlyPort))
-	if err != nil {
-		return &resp, err
+	// Poll pod information and search for the pod with the container. If
+	// the pod is not found, and there are pods with containers that aren't
+	// fully initialized, delay for a little bit and try again.
+	for attempt := 1; ; attempt++ {
+		podInfo, err := p.getPodInfoFromInsecureKubeletPort()
+		if err != nil {
+			return &resp, err
+		}
+
+		notAllContainersReady := false
+		for _, item := range podInfo.Items {
+			switch lookUpContainerInPod(containerID, item.Status) {
+			case containerInPod:
+				resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("sa:%v", item.Spec.ServiceAccountName)})
+				resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("ns:%v", item.Metadata.Namespace)})
```

Comment:
We will definitely be wanting to add more selectors here in the near future. Would probably be good to place this in a dedicated function so the case logic doesn't get huge when we do.

Reply:
Awesome, will do.